testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/core/basic/CacheControlAnnotationTest.java

```diff
@@ -108,4 +113,20 @@ public void testResourceCompositeNoCache() {
         response.close();
     }
 
+    /**
+     * @tpTestDetails Test for correct value of max-age of inherited cache annotation
+     * @tpSince RESTEasy 6.3.0
+     */
+    @Test
+    public void testInheritedResourceValid() {
+        WebTarget base = client.target(generateURL("/inheritance"));
+        Response response = base.request().get();
+
+        Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+        CacheControl cc = CacheControl.valueOf(response.getHeaderString("cache-control"));
+        Assert.assertFalse("Cache should not be private", cc.isPrivate());
+        Assert.assertEquals("Wrong age of cache", 3600, cc.getMaxAge());
+
+        response.close();
```

Comment:
May as well use try-with-resources since we will likely not back port this change.